File: java-adapter/src/main/java/preadapter/service/AnalyzeAST.java
package preadapter.service;

import com.github.javaparser.StaticJavaParser;
import com.github.javaparser.ast.CompilationUnit;
import preadapter.Logger;
import preadapter.domain.ASTVisitorDebug;
import preadapter.domain.Datastructure;
import preadapter.domain.ASTVisitor;
import preadapter.infrastructure.CreateYML;

import java.io.FileNotFoundException;
import java.util.List;
import java.io.File;

public class AnalyzeAST {

    public void SearchRecurseCSharpToYAML(String fromDir, String toDir, Logger logger) {
        Datastructure datastructure = new Datastructure(logger);
        var recurseFileProcess = new RecursiveFileProcessor();
        List<String> listFilePaths = recurseFileProcess.SearchPath(fromDir);
        for (String filePath : listFilePaths) {
            if (filePath.endsWith(".java")) {
                logger.logInfo(String.format("Analyzing %s", filePath));
                CompilationUnit cu;
                try {
                    cu = StaticJavaParser.parse(new File(filePath));
                    new ASTVisitor(datastructure, filePath, logger).visit(cu, null);
                } catch (FileNotFoundException e) {
                    e.printStackTrace();
                }
            }
        }
        datastructure.ResolveClassNames();
        String toFile = toDir + "/output.yml";
        new CreateYML().Create(datastructure, toFile, logger);
        //new Infrastructure.CreateYml().Create(datastructure, toDir, logger);
    }
}

File: dotnet-adapter/DotnetPreAdapter/service/AnalyzeAST.cs
using System;
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.CSharp;
using Microsoft.CodeAnalysis.CSharp.Syntax;
using System.Text.RegularExpressions;

namespace DotNetPreAdapter
{
    namespace Service
    {
        public class AnalyzeAST
        {

            public void SearchRecurseCSharpToYAML(string fromDir, string toDir, Utils.Logger logger)
            {
                Domain.Datastructure datastructure = new Domain.Datastructure(logger);
                var recurseFileProcess = new RecursiveFileProcessor();
                List<string> listFilePaths = recurseFileProcess.SearchPath(fromDir);
                foreach (string filePath in listFilePaths)
                {
                    if (filePath.EndsWith(".cs"))
                    {
                        logger.LogInfo($"Analyzing {filePath}");
                        string programText = File.ReadAllText(filePath);
                        SyntaxTree tree = CSharpSyntaxTree.ParseText(programText);
                        CompilationUnitSyntax root = tree.GetCompilationUnitRoot();

                        var compilation = CSharpCompilation.Create("HelloWorld")
                            .AddReferences(MetadataReference.CreateFromFile(
                                typeof(string).Assembly.Location)).AddSyntaxTrees(tree);
                        SemanticModel model = compilation.GetSemanticModel(tree);

                        string reducedFilePath = Regex.Replace(filePath.Replace(fromDir, ""), "^[\\s*\\/]*", "");

                        var visitor = new Domain.ASTVisitor(datastructure, reducedFilePath, model, logger);
                        visitor.Visit(root);
                    }

                }
                datastructure.ResolveClassNames();
                new Infrastructure.CreateYml().Create(datastructure, toDir, logger);

            }
        }
    }
}

